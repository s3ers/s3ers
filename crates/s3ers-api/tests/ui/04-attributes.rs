use s3ers_api::s3ers_api;

s3ers_api! {
    metadata: {
        description: "Does something.",
        method: POST, // An `http::Method` constant. No imports required.
        name: "some_endpoint",
        path: "/some/endpoint/:baz",
        authentication: None,
    }

    #[not_a_real_attribute_should_fail]
    request: {
        pub foo: String,
        #[s3ers_api(header = CONTENT_TYPE)]
        pub content_type: String,
        #[s3ers_api(query)]
        pub bar: String,
        #[s3ers_api(path)]
        pub baz: String,
    }

    response: {
        pub value: String,
    }
}

fn main() {}
