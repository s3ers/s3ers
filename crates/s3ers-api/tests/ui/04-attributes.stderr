error: cannot find attribute `not_a_real_attribute_should_fail` in this scope
  --> $DIR/04-attributes.rs:12:7
   |
12 |     #[not_a_real_attribute_should_fail]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
